codec/src/main/java/io/netty/handler/codec/compression/JdkZlibEncoder.java

```diff
@@ -35,6 +42,84 @@
  */
 public class JdkZlibEncoder extends ZlibEncoder {
 
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(JdkZlibEncoder.class);
+    private static final Method DEFLATER_SET_INPUT_BYTEBUFFER;
+    private static final Method DEFLATER_DEFLATE_BYTEBUFFER;
+    private static final Method CRC32_UPDATE_BYTEBUFFER;
+
+    static {
+        Method deflaterSetInput = null;
+        Method deflaterDeflate = null;
+        Method crc32Update = null;
+        if (PlatformDependent.javaVersion() >= 11) {
+            // discover deflater.setInput(ByteBuffer)
+            Object maybeException =
+                    AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                        @Override
+                        public Object run() {
+                            try {
+                                return Deflater.class.getDeclaredMethod("setInput", ByteBuffer.class);
+                            } catch (NoSuchMethodException e) {
+                                return e;
+                            } catch (SecurityException e) {
+                                return e;
+                            }
+                        }
+                    });
+            if (maybeException instanceof Method) {
+                logger.debug("Deflater.setInput(ByteBuffer): available");
+                deflaterSetInput = (Method) maybeException;
+            } else {
+                logger.debug("Deflater.setInput(ByteBuffer): unavailable");
+            }
+
+            // discover deflater.deflate(ByteBuffer, int)
+            maybeException =
+                    AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                        @Override
+                        public Object run() {
+                            try {
+                                return Deflater.class.getDeclaredMethod("deflate", ByteBuffer.class, int.class);
+                            } catch (NoSuchMethodException e) {
+                                return e;
+                            } catch (SecurityException e) {
+                                return e;
+                            }
+                        }
+                    });
+            if (maybeException instanceof Method) {
+                logger.debug("Deflater.deflate(ByteBuffer, int): available");
+                deflaterDeflate = (Method) maybeException;
+            } else {
+                logger.debug("Deflater.deflate(ByteBuffer, int): unavailable");
+            }
+
+            // discover crc32.update(ByteBuffer)
```

Comment:
Don't we already have this in `ByteBufChecksum` ?

Reply:
Now I know. Using the one from `ByteBufChecksum` now.